### DISCUSSION TITLE
100% Normalization Uncertainty

### DISCUSSION THREAD

**Discussion by Tomoya-Iizawa** [scikit-hep/cabinetry#462] [Q&A] [answered]
Hi,

I am trying to include systematics with 100% normalization uncertainty. For some sample, it works with "Normalization: 1.0", but the sample I would like to implement, fit fails. But it even works with "Normalization: 0.999999", so is there some possible reason of the fail? One thing is that this sample has very low stat, can this be reason?

Best,
Tomoya

**Comment by alexander-held** [comment, marked as answer]
Hi Tomoya, this happens due to the algorithm being used to perform extrapolations. For normalization uncertainties, the extrapolation is exponential (["code 4" in `pyhf`](https://pyhf.readthedocs.io/en/v0.7.6/_generated/pyhf.interpolators.code4.html#pyhf.interpolators.code4)). This has the desirable property of never predicting negative yields as long as the yields at the up/down templates are larger than 0. If you have a variation that can decrease the normalization of a sample by 100%, the extrapolation breaks (technically I think there will be a division by zero causing issues).

What I am used to seeing in practice are solutions where the variation in down direction is set to something very close to 100%. You can then still pull the nuisance parameter beyond the [-1, 1] interval and get physical values and avoid breaking anything.

I have not thought further about whether it would make sense to define this 100% variation as a special case with an algorithm that can handle a prediction of exactly 0 events at the -100% variation and a constant 0 events beyond that. The discontinuity in first derivative of the model prediction at that nuisance parameter point might not be a big issue in practice, but the algorithm would presumably be somewhat ad hoc for no obvious (to me, at the moment at least) gain. However, now thinking more about this, it would probably be useful to catch specifically the 100% normalization uncertainty case in `pyhf` with a dedicated error message.

**Reply by Tomoya-Iizawa** [reply]
I see, thank you! so I will put something as Normalization: 0.999999.
